### PR TITLE
Record access token usage and send sentry if issue arises

### PR DESCRIPTION
**Pull request by LunarX**
Uses the new AccessTokenUsageInterceptor to record issue with access token to Sentry

Depends on https://github.com/Infomaniak/android-core/pull/185

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/MainApplication.kt [hunk 1]
```diff
@@ -173,8 +184,8 @@ class MainApplication : Application(), ImageLoaderFactory, DefaultLifecycleObser
             refreshTokenError(AccountUtils.currentUser!!)
         }
 
-        override suspend fun getApiToken(): ApiToken {
-            return AccountUtils.currentUser!!.apiToken
+        override suspend fun getApiToken(): ApiToken? {
+            return AccountUtils.currentUser?.apiToken
         }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/MainApplication.kt:189]:
```suggestion
        override suspend fun getApiToken(): ApiToken? = AccountUtils.currentUser?.apiToken
```

File: app/src/main/java/com/infomaniak/drive/MainApplication.kt [hunk 2]
```diff
@@ -129,7 +132,15 @@ class MainApplication : Application(), ImageLoaderFactory, DefaultLifecycleObser
...
         AccountUtils.onRefreshTokenError = refreshTokenError
         initNotificationChannel()
-        HttpClient.init(tokenInterceptorListener())
+        val tokenInterceptorListener = tokenInterceptorListener()
+        HttpClientConfig.customInterceptors = listOf(
+            AccessTokenUsageInterceptor(
+                tokenInterceptorListener = tokenInterceptorListener,
+                previousApiCall = uiSettings.accessTokenApiCallRecord,
+                updateLastApiCall = { uiSettings.accessTokenApiCallRecord = it },
+            )
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/MainApplication.kt:141]:
```suggestion
            ),
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Review by tevincent — CHANGES_REQUESTED** [review]
Same as kevin

File: app/src/main/java/com/infomaniak/drive/MainApplication.kt [hunk 3]
```diff
@@ -77,7 +79,8 @@ class MainApplication : Application(), ImageLoaderFactory, DefaultLifecycleObser
 
         ProcessLifecycleOwner.get().lifecycle.addObserver(this)
 
-        AppCompatDelegate.setDefaultNightMode(UiSettings(this).nightMode)
+        val uiSettings = UiSettings(this)
+        AppCompatDelegate.setDefaultNightMode(uiSettings.nightMode)
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/drive/MainApplication.kt:83]: Was that really necessary? Creating a `val` just for that is a bit too much?

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/drive/MainApplication.kt:83]: It's used later in the same function, lines 139 & 140.

**Review by KevinBoulongne — APPROVED** [review]

File: app/src/main/java/com/infomaniak/drive/MainApplication.kt (hunk 3 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/drive/MainApplication.kt:83]: Maybe this function is a little too long 🤔

**Review by tevincent — APPROVED** [review]

**Pull request merged**